src/main/java/ch/njol/skript/lang/parser/ParserInstance.java

```diff
@@ -321,6 +324,40 @@ public final boolean isCurrentSection(Class<? extends TriggerSection>... section
 		}
 		return false;
 	}
+	
+	// Loop API
+	
+	private final List<Loopable> currentLoops = new ArrayList<>();
+	
+	/**
+	 * @param loops A new list of loops to handle.
+	 */
+	public void setCurrentLoops(@NotNull List<Loopable> loops) {
+		currentLoops.clear();
+		currentLoops.addAll(loops);
+	}
+	
+	/**
+	 * @return A list of all loops this ParserInstance is currently within.
+	 */
+	public @NotNull @Unmodifiable List<Loopable> getCurrentLoops() {
+		return Collections.unmodifiableList(currentLoops);
+	}
+	
+	/**
+	 * Adds the loop to the current loops list.
+	 * @param loopable The loop to add.
+	 */
+	public void pushLoop(Loopable loopable) {
+		currentLoops.add(loopable);
+	}
+	
+	/**
+	 * Removes the last loop from the current loops list.
+	 */
+	public void popLoop() {
+		currentLoops.remove(currentLoops.size() - 1);
+	}
```

Comment:
Very unsure about this, wouldn't be needed if there was a way to get the current effect from an expression. Have I missed something?